modules/core/arrow-test/src/main/kotlin/arrow/test/laws/Law.kt

```diff
@@ -1,81 +1,62 @@
 package arrow.test.laws
 
+import arrow.test.generators.genTuple
 import arrow.typeclasses.Eq
-import io.kotlintest.matchers.should
+import io.kotlintest.TestContext
 import io.kotlintest.properties.Gen
+import io.kotlintest.should
 
 fun throwableEq() = Eq { a: Throwable, b ->
   a::class == b::class && a.message == b.message
 }
 
-data class Law(val name: String, val test: () -> Unit)
+data class Law(val name: String, val test: TestContext.() -> Unit)
 
 fun <A> A.equalUnderTheLaw(b: A, eq: Eq<A>): Boolean =
   eq.run { eqv(b) }
 
-fun <A> A.shouldBe(b: A, eq: Eq<A>): Unit = eq.run {
+fun <A> A.shouldBeEq(b: A, eq: Eq<A>): Unit = eq.run {
```

Comment:
.... why aren't we using this everywhere?

Reply:
`forAll` needs a `Boolean` as a return type for its function the `shouldBeEq` returns a `Unit`